sim/core/apl_actions_misc.go

```diff
@@ -238,15 +238,17 @@ func (action *APLActionItemSwap) IsReady(sim *Simulation) bool {
 	return (action.swapSet == proto.APLActionItemSwap_Main) == action.character.ItemSwap.IsSwapped()
 }
 func (action *APLActionItemSwap) Execute(sim *Simulation) {
-	if sim.Log != nil {
-		action.character.Log(sim, "Item Swap to set %s", action.swapSet)
-	}
-
-	if action.swapSet == proto.APLActionItemSwap_Main {
-		action.character.ItemSwap.reset(sim)
+	if action.character.ItemSwap.swapSet == action.swapSet {
+		if sim.Log != nil {
+			action.character.Log(sim, "Item Swap already set to %s", action.swapSet)
+		}
```

Comment:
Should we just break after this?

Reply:
We could, however the `SwapItems` call will do the check as well and that way we have that logic all in one place